app/src/screens/MapScreen.js

```diff
@@ -97,97 +111,48 @@ function MapScreen(): React.Node {
   });
 
+  const rangeFactor = useMemo(
+    () =>
+      (-1 * 255) /
+      (maxRssiToDisplay ?? maxRssi - (minRssiToDisplay ?? minRssi)),
+    [maxRssiToDisplay, minRssiToDisplay],
+  );
+
   let minRssi = -1000;
   let maxRssi = 0;
-  let bestLongitude;
-  let bestLatitude;
-  let bestElevation;
-  let bestBearing;
   const fileInput = useRef(null);
 
   function handleOpenClick() {
     fileInput.current && fileInput.current.click && fileInput.current.click();
   }
 
   function handleFile(e: SyntheticInputEvent<HTMLInputElement>) {
-    let newLayers: Array<Point> = [];
-    let xyPoints: Array<XYPoint> = [];
-    setLayers(null);
-    setGraphData(null);
+    const newLayers: LayerDict = {};
     const files = e.target.files;
 
     Array.from(files).forEach((file: File) => {
+      const name = file.name;
       const reader = new FileReader();
       reader.onload = _e => {
         const content = reader.result;
         if (content !== null && typeof content === 'string') {
           const lines = processFileData(content);
-          newLayers = newLayers.concat(lines);
-          xyPoints = xyPoints.concat(pointsToXY(lines));
-          const rangeFactor = (-1 * 255) / (maxRssi - minRssi);
-
-          const iconLayer = new IconLayer<Point>({
-            id: 'icon-layer',
-            data: newLayers,
-            pickable: true,
-            // iconAtlas and iconMapping are required
-            iconAtlas: 'arrow.png',
-            iconMapping: ICON_MAPPING,
-            // getIcon: return a string
-            getIcon: (_d: Point) => 'marker',
-            sizeScale: 2,
-            getPosition: (d: Point) => [d.latitude, d.longitude, d.height],
-            getSize: (_d: Point) => 15,
-            getColor: (d: Point) => {
-              const red = parseInt(255 - (minRssi - d.rssi) * rangeFactor);
-              const blue = 255 - red;
-              const green = 255 - red - blue;
-              return [red, green, blue, 255];
-            },
-            getAngle: (d: Point) => 180 - d.bearing,
-            billboard: false,
-            onHover: info => setHoverInfo(info),
-          });
-
-          const highestSignalLayer = new PointCloudLayer({
-            id: 'highestSignal',
-            data: [
-              {
-                position: [bestLatitude, bestLongitude, bestElevation],
-                message:
-                  bestLatitude +
-                  ', ' +
-                  bestLongitude +
-                  ' ' +
-                  bestElevation +
-                  ' meters ' +
-                  minRssi +
-                  'dBm ' +
-                  bestBearing +
-                  '\u00b0',
-              },
-            ],
-            pickable: true,
-            coordinateSystem: COORDINATE_SYSTEM.LNGLAT,
-            coordinateOrigin: [newLayers[0].latitude, newLayers[0].longitude],
-            pointSize: 10,
+          const xyPoints = pointsToXY(lines);
+          newLayers[name] = {
```

Comment:
kind of an edge case, but what if you upload 2 files with the same name?  won't this overwrite it?

for purposes of demo this probably doesn't matter though.

Reply:
Yup!
I considered this as well, but it's kind of annoying to handle, and figured we can fix it later.